src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/applicationmanager.cpp

```diff
@@ -143,22 +143,26 @@ APPLICATION_MANAGER::RecycleApplicationFromManager(
             }
         }
 
-        // If we receive a request at this point.
-        // OutOfProcess: we will create a new application with new configuration
-        // InProcess: the request would have to be rejected, as we are about to call g_HttpServer->RecycleProcess
-        // on the worker process
-
         if (!applicationsToRecycle.empty())
         {
             for (auto& application : applicationsToRecycle)
             {
                 try
                 {
-                    application->ShutDownApplication(/* fServerInitiated */ false);
+                    if (GetShutdownDelay() == std::chrono::milliseconds::zero())
```

Comment:
Can you say a bit more about when this happens?  Is the user setting zero explicitly to revert to the old behavior?

Reply:
Maybe extract a local `useLegacyShutdownProcess = GetShutdownDelay() == std::chrono::milliseconds::zero()` (name TBD)?

Reply:
Yeah, this is when the user sets 0 explicitly. It will default to 0 if/when we backport to 8.0 so that it's opt-in instead of opt-out.